modules/shapefile/src/lib/parsers/parse-shapefile.js

```diff
@@ -5,26 +5,44 @@ import {zipBatchIterators} from '../streaming/zip-batch-iterators';
 import {SHPLoader} from '../../shp-loader';
 import {DBFLoader} from '../../dbf-loader';
 
+// eslint-disable-next-line max-statements
 export async function* parseShapefileInBatches(asyncIterator, options, context) {
+  const {metadata} = options;
   const {reproject = false, _targetCrs = 'WGS84'} = (options && options.gis) || {};
   const {parseInBatches, fetch, url} = context;
   const {shx, cpg, prj} = await loadShapefileSidecarFiles(options, context);
 
   // parse geometries
-  const shapeIterator = await parseInBatches(asyncIterator, SHPLoader, options); // {shp: shx}
-  const shapeHeader = (await shapeIterator.next()).value;
+  const shapeIterator = await parseInBatches(asyncIterator, SHPLoader, options);
 
   // parse properties
   let propertyIterator;
-  let iterator;
   const dbfResponse = await fetch(replaceExtension(url, 'dbf'));
   if (dbfResponse.ok) {
     propertyIterator = await parseInBatches(dbfResponse, DBFLoader, {
+      ...options,
```

Comment:
I assume it's preferable to pass `options` to sub loaders, i.e. if the user wants to override `fetch`